cmd/osm-injector/osm-injector.go

```diff
@@ -0,0 +1,240 @@
+	flags.StringVar(&vaultOptions.VaultHost, "vault-host", "vault.default.svc.cluster.local", "Host name of the Hashi Vault")
+	flags.StringVar(&vaultOptions.VaultToken, "vault-token", "", "Secret token for the the Hashi Vault")
+	flags.StringVar(&vaultOptions.VaultRole, "vault-role", "openservicemesh", "Name of the Vault role dedicated to Open Service Mesh")
+	flags.IntVar(&vaultOptions.VaultPort, "vault-port", 8200, "Port of the Hashi Vault")
+
+	// Cert-manager certificate manager/provider options
+	flags.StringVar(&certManagerOptions.IssuerName, "cert-manager-issuer-name", "osm-ca", "cert-manager issuer name")
+	flags.StringVar(&certManagerOptions.IssuerKind, "cert-manager-issuer-kind", "Issuer", "cert-manager issuer kind")
+	flags.StringVar(&certManagerOptions.IssuerGroup, "cert-manager-issuer-group", "cert-manager.io", "cert-manager issuer group")
+
+	_ = clientgoscheme.AddToScheme(scheme)
+	_ = v1beta1.AddToScheme(scheme)
+}
+
+func main() {
+	log.Info().Msgf("Starting osm-injector %s; %s; %s", version.Version, version.GitCommit, version.BuildDate)
+	if err := parseFlags(); err != nil {
+		log.Fatal().Err(err).Msg("Error parsing cmd line arguments")
+	}
+	if err := logger.SetLogLevel(verbosity); err != nil {
+		log.Fatal().Err(err).Msg("Error setting log level")
+	}
+
+	// Initialize kube config and client
+	kubeConfig, err := clientcmd.BuildConfigFromFlags("", kubeConfigFile)
+	if err != nil {
+		log.Fatal().Err(err).Msgf("Error creating kube config (kubeconfig=%s)", kubeConfigFile)
+	}
+	kubeClient := kubernetes.NewForConfigOrDie(kubeConfig)
+
+	// Initialize the generic Kubernetes event recorder and associate it with the osm-injector pod resource
+	injectorPod, err := getInjectorPod(kubeClient)
+	if err != nil {
+		log.Fatal().Msg("Error fetching osm-injector pod")
+	}
+	eventRecorder := events.GenericEventRecorder()
+	if err := eventRecorder.Initialize(injectorPod, kubeClient, osmNamespace); err != nil {
+		log.Fatal().Msg("Error initializing generic event recorder")
+	}
+
+	// This ensures CLI parameters (and dependent values) are correct.
+	if err := validateCLIParams(); err != nil {
+		events.GenericEventRecorder().FatalEvent(err, events.InvalidCLIParameters, "Error validating CLI parameters")
+	}
+
+	stop := signals.RegisterExitHandlers()
+	_, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// Start the default metrics store
+	metricsstore.DefaultMetricsStore.Start()
+
+	// Initialize Configurator to watch osm-config ConfigMap
+	cfg := configurator.NewConfigurator(kubeClient, stop, osmNamespace, osmConfigMapName)
+	configMap, err := cfg.GetConfigMap()
+	if err != nil {
+		log.Error().Err(err).Msgf("Error parsing ConfigMap %s", osmConfigMapName)
+	}
+	log.Info().Msgf("Initial ConfigMap %s: %s", osmConfigMapName, string(configMap))
+
+	// Initialize kubernetes.Controller to watch kubernetes resources
+	kubeController, err := k8s.NewKubernetesController(kubeClient, meshName, stop)
```

Comment:
I made a quick search and it feels to me this guy is overkill, and we should strip it down from injector in the future.

This seems to only be needed to issue a couple GetNamespace / IsMonitoredNamespace, whereas it will be synchronizing services, pods, svcAccounts and endpoints, and doesn't seem to really need any of those.

Reply:
I agree, any ideas on how to decouple informer caches? If we want to use limited informer set, maybe we can allow clients to specify which ones.